floor_generator/lib/processor/query_processor.dart

```diff
@@ -0,0 +1,145 @@
+import 'package:floor_generator/misc/extension/dart_type_extension.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:floor_generator/processor/error/query_processor_error.dart';
+import 'package:floor_generator/processor/processor.dart';
+import 'package:floor_generator/value_object/query.dart';
+
+class QueryProcessor extends Processor<Query> {
+  final QueryProcessorError _processorError;
+
+  final String _query;
+
+  final List<ParameterElement> _parameters;
+
+  QueryProcessor(MethodElement methodElement, this._query)
+      : _parameters = methodElement.parameters,
+        _processorError = QueryProcessorError(methodElement);
+
+  @override
+  Query process() {
+    _assertNoNullableParameters();
+
+    final listParameters = <ListParameter>[];
+    final newQuery = _processParameters(listParameters);
```

Comment:
💡 Passing the `listParameters` list into `_processParameters` makes it a bit trickier to follow what's happening to the `listParameters`. Instead, we could make `_processParameters` return a structure that holds both the returned query and the `listParameters`, which is basically `Query`. Thereby, inlining `_processParameters` might help to improve readability.

Reply:
good point! I'll do that.